scripts/unicode.py

```diff
@@ -106,6 +122,39 @@ def _load_unicode_data(self):
             if category == 'M' or 'M' in expanded_categories.get(category, []):
                 self.general_category_mark.append(char_int)
 
+    def _load_standardized_variants(self):
+        for line in self._fetch("StandardizedVariants.txt").splitlines():
+            strip_comments = line.split('#', 1)[0].strip()
+            if not strip_comments:
+                continue
+
+            pieces = strip_comments.split(';')
+            assert len(pieces) == 3
+
+            variation_sequence, description, differences = pieces[0], pieces[1].strip(), pieces[2]
+
+            # Don't use variations that only apply in particular shaping environments.
+            if differences:
+                continue
+
+            # Look for entries where the description field is a codepoint name.
+            if description in self.name_to_char_int:
+                char_int = self.name_to_char_int[description]
```

Comment:
we'll only enter this branch for the CJK compatibility ideographs, right?

this is related to my previous general comment, but perhaps we should specialize this process to just be about CJK compatibility? it might be worth being clear that stuff like mathematical symbols or myanmar are unaffected by this transformation.

so, we'd specifically look for the CJK description in this table, name the generated table something CJK related, and then call the process something CJK related as well (instead of the more generic `svar`)

i'm curious what other libraries do for this, though.

Reply:
Currently, the CJK compatibility codepoints are the only entires in StandardizedVariants.txt which have description fields that match codepoint names, which indicate direct correspondences. I imagine it's possible Unicode could use the same mechanism for other sequences in the future, which is what led me to write the code generic and not specialized for CJK. That said, I don't know how likely Unicode is to do this, so if you think it's best to limit the code to the CJK sequences, I can do that.

Reply:
Also, I don't know of any other libraries which implement this. My impression is that this feature is simply not widely known; it seems like it should be useful in a lot of situations, such as where one needs NFC text but doesn't need to match the standard `toNFC` algorithm, and one wishes to avoid losing information to the extent possible.

Reply:
yeah, I think this is an obscure enough feature that we'd want to be very specific in what does. nice work btw on the iterator docstring -- that should suffice for helping people figure out the new concepts involved.

Reply:
Sounds good; I've now made this change; `svar` is now `cjk_compat_variants` and associated comments and identifiers are updated.